api/download_test.go

```diff
@@ -84,11 +84,25 @@ func TestHandleFileNotPublished(t *testing.T) {
 			fetchMetadata := func(path string) (files.Metadata, error) {return files.Metadata{State: tt.state}, nil}
 			downloadFile := func(path string) (io.ReadCloser, error) {return nil, nil}
 
-			h := api.CreateV1DownloadHandler(fetchMetadata, downloadFile)
+			h := api.CreateV1DownloadHandler(fetchMetadata, downloadFile, "")
 
 			h.ServeHTTP(rec, req)
 
 			assert.Equalf(t, tt.expectedStatus, rec.status, "CreateV1DownloadHandler(%v)", tt.name)
 		})
 	}
 }
+
+func TestHandlingErrorWhenBadPublicBucketUrl(t *testing.T) {
```

Comment:
is the intent to test unhappy path at this level and happy path at the component-test level?

Reply:
yes, but we've moved the error checking to the config now, so it will check on app startup instead